### PR TITLE
votca-*: merge packages into one

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

**Comment by alalazo** [comment]
Hi @junghans If I read the PR correctly the `votca-*` packages are merged into a single `votca` package. Does this mean that the old packages become deprecated and from now on only the "single" package will be maintained?

**Comment by junghans** [comment]
> Hi @junghans If I read the PR correctly the `votca-*` packages are merged into a single `votca` package. Does this mean that the old packages become deprecated and from now on only the "single" package will be maintained?

Correct! But the old versions aren’t available as a mono build, so I would like to keep them for now.

File: var/spack/repos/builtin/packages/votca-csg-tutorials/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/votca-csg-tutorials/package.py:23]: We should first deprecate these versions and remove them in a future Spack release, see https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions

File: var/spack/repos/builtin/packages/votca/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/votca/package.py:43]: Are these Python libs not needed at run-time?

Comment (junghans) [reply var/spack/repos/builtin/packages/votca/package.py:43]: No, CMake looks for them and otherwise disables some features.

File: var/spack/repos/builtin/packages/votca-csg-tutorials/package.py
Comment (junghans) [reply var/spack/repos/builtin/packages/votca-csg-tutorials/package.py:23]: Good point, will add `deprecated=True`.

Comment (junghans) [reply var/spack/repos/builtin/packages/votca-csg-tutorials/package.py:23]: But I removed `master` as the code was removed from the branch and it won't build either way.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/votca-csg-tutorials/package.py:23]: I would still deprecate master/stable in case someone already has those branches checked out locally in their downloads cache.

Comment (junghans) [reply var/spack/repos/builtin/packages/votca-csg-tutorials/package.py:23]: Hmm, I feel the chance that somebody trying to build a deprecated `master` and ending up with obscure error messages as the branch has no code on it, is much bigger than somebody trying to build from their downloads cache.

In the end I am happy to follow the packing guide, but in this case this seems like we are trying to help 0.1% that have a local cache and screwing over 99.9% that don't read the deprecation message.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/votca-csg-tutorials/package.py:23]: > don't read the deprecation message.

Note that the deprecation message is a prompt that asks you if you're sure you want to install this version despite it being deprecated, so it's a little harder to ignore. You can still miss it if you use `spack install --yes-to-all` though.

You're the maintainer of these packages, so I'll leave the final decision to you. These are more "best practices" than strict policies.

Comment (junghans) [reply var/spack/repos/builtin/packages/votca-csg-tutorials/package.py:23]: Yeah, I know my users, so I will opt for dropping master ;-)

File: var/spack/repos/builtin/packages/votca/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/votca/package.py:27]: I wonder how useful this variant actually is. I think it's relatively easy to express this in Spack's spec syntax. For example, the following:
```console
$ spack install votca+new-gmx
$ spack install votca~new-gmx
```
is equivalent to:
```console
$ spack install votca^gromacs@2020:
$ spack install votca^gromacs@:2019
```

Comment (junghans) [reply var/spack/repos/builtin/packages/votca/package.py:27]: I know my users, the latter might be a bit too complicated for most. I want it to work by default and that means gromacs-2019 is required. If gromacs ever re-implements tabulated interactions we can drop it again. (see https://gitlab.com/gromacs/gromacs/-/issues/1347)

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
